Sources/SwiftGenKit/Parsers/Strings/StringsDictEntry.swift

```diff
@@ -0,0 +1,169 @@
+    let variableWidthRules = try container.decodeIfPresent([String: String].self, forKey: .variableWidthRules)
+
+    // We either have a formatKey OR we have a variableWidthRule.
+    switch (formatKey, variableWidthRules) {
+    case (.none, .none), (.some, .some):
+      throw Strings.ParserError.invalidFormat
+    case let (.some(formatKey), .none):
+      let variables = try StringsDict.decodeVariableRules(in: container, formatKey: formatKey)
+      self = .pluralEntry(PluralEntry(formatKey: formatKey, variables: variables))
+    case let (.none, .some(variableWidthRules)):
+      self = .variableWidthEntry(VariableWidthEntry(rules: variableWidthRules))
+    }
+  }
+
+  private static func decodeVariableRules(
+    in container: KeyedDecodingContainer<StringsDict.CodingKeys>,
+    formatKey: String
+  ) throws -> [String: PluralEntry.VariableRule] {
+    var variables = [String: PluralEntry.VariableRule]()
+
+    for variableKey in try StringsDict.variableKeysFromFormatKey(formatKey) {
+      let variable = try container.decode(PluralEntry.VariableRule.self, forKey: .make(key: variableKey))
+      variables[variableKey] = variable
+
+      // Nested FormatKey in Variable. Check if one of the strings (zero, one, ...) contains format keys,
+      // decode them and add them to `variables`
+      let childVariableKeys = Set(try variable.formatStrings.flatMap { try StringsDict.variableKeysFromFormatKey($0) })
+      for variableKey in Array(childVariableKeys) {
+        variables[variableKey] = try container.decode(PluralEntry.VariableRule.self, forKey: .make(key: variableKey))
+      }
+    }
+    return variables
+  }
+
+  /// Parses a format string and returns an array of discovered variable keys.
+  /// Every variable key that is contained within the format string is preceded
+  /// by the %#@ characters and followed by the @ character.
+  private static func variableKeysFromFormatKey(_ formatKey: String) throws -> [String] {
+    let pattern = #"%(?>\d\$)?#@([\w\.\p{Pd}]+)@"#
+    let regex = try NSRegularExpression(pattern: pattern, options: [])
+    let nsrange = NSRange(formatKey.startIndex..<formatKey.endIndex, in: formatKey)
+    let matches = regex.matches(in: formatKey, options: [], range: nsrange)
+
+    return matches.compactMap { match -> String? in
+      let captureGroupRange = match.range(at: 1)
+      guard
+        captureGroupRange.location != NSNotFound,
+        let range = Range(captureGroupRange, in: formatKey)
+      else {
+        return nil
+      }
+
+      return String(formatKey[range])
+    }
+  }
+}
+
+// - MARK: Helpers
+
+extension StringsDict.PluralEntry {
+  var translation: String? {
+    var result = formatKey
+    variables.forEach { keyValuePair in
+      let (key, variable) = keyValuePair
+      result = result.replacingOccurrences(of: "%#@\(key)@", with: variable.other)
+    }
+
+    return "Plural case 'other': \(result)"
```

Comment:
I'm not sure what this is for and why it's using the case for `other` specifically? What if you have a plural translation that only has translations for `one` and `many` but a nil value (no entry) for `other`?

Reply:
According to the [Stringsdict File Format](https://developer.apple.com/library/archive/documentation/MacOSX/Conceptual/BPInternational/StringsdictFileFormat/StringsdictFileFormat.html), the `other` case is required, that's why it is the only non-optional `VariableRule`.

Reply:
Ah, missed that, good point 👍